In [60]:
import os
os.getcwd()
!pip install scipy

from scipy.stats import f_oneway
from scipy.stats import ttest_ind

Defaulting to user installation because normal site-packages is not writeable


In [44]:
df = pd.read_csv("cars_dataset.csv",encoding='latin1')

<IPython.core.display.Javascript object>

In [45]:
df.head()

Sales_ID                          name  year  selling_price  km_driven  \
0         1        Maruti Swift Dzire VDI  2014         450000     145500   
1         2  Skoda Rapid 1.5 TDI Ambition  2014         370000     120000   
2         3      Honda City 2017-2020 EXi  2006         158000     140000   
3         4     Hyundai i20 Sportz Diesel  2010         225000     127000   
4         5        Maruti Swift VXI BSIII  2007         130000     120000   

     fuel seller_type transmission         owner     mileage   engine  \
0  Diesel  Individual       Manual   First Owner   23.4 kmpl  1248 CC   
1  Diesel  Individual       Manual  Second Owner  21.14 kmpl  1498 CC   
2  Petrol  Individual       Manual   Third Owner   17.7 kmpl  1497 CC   
3  Diesel  Individual       Manual   First Owner   23.0 kmpl  1396 CC   
4  Petrol  Individual       Manual   First Owner   16.1 kmpl  1298 CC   

    max_power  seats      City  Postal_code Sales_status  
0      74 bhp      5    Mumbai       400001            Y  
1  103.52 bhp      5    Ranchi       834008            Y  
2      78 bhp      5     Noida       201301            Y  
3      90 bhp      5  Dehradun       248141            Y  
4    88.2 bhp      5   Chennai       600020            Y

In [46]:
df.shape

(7906, 16)

In [47]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7906 entries, 0 to 7905
Data columns (total 16 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Sales_ID       7906 non-null   int64 
 1   name           7906 non-null   object
 2   year           7906 non-null   int64 
 3   selling_price  7906 non-null   int64 
 4   km_driven      7906 non-null   int64 
 5   fuel           7906 non-null   object
 6   seller_type    7906 non-null   object
 7   transmission   7906 non-null   object
 8   owner          7906 non-null   object
 9   mileage        7906 non-null   object
 10  engine         7906 non-null   object
 11  max_power      7906 non-null   object
 12  seats          7906 non-null   int64 
 13  City           7906 non-null   object
 14  Postal_code    7906 non-null   int64 
 15  Sales_status   7906 non-null   object
dtypes: int64(6), object(10)
memory usage: 988.4+ KB


In [48]:
df.isnull().sum()

Sales_ID         0
name             0
year             0
selling_price    0
km_driven        0
fuel             0
seller_type      0
transmission     0
owner            0
mileage          0
engine           0
max_power        0
seats            0
City             0
Postal_code      0
Sales_status     0
dtype: int64

In [49]:
df.describe()

Sales_ID         year  selling_price     km_driven        seats  \
count  7906.000000  7906.000000   7.906000e+03  7.906000e+03  7906.000000   
mean   4070.106248  2013.983936   6.498137e+05  6.918866e+04     5.416393   
std    2345.770159     3.863695   8.135827e+05  5.679230e+04     0.959208   
min       1.000000  1994.000000   2.999900e+04  1.000000e+00     2.000000   
25%    2037.250000  2012.000000   2.700000e+05  3.500000e+04     5.000000   
50%    4076.500000  2015.000000   4.500000e+05  6.000000e+04     5.000000   
75%    6102.000000  2017.000000   6.900000e+05  9.542500e+04     5.000000   
max    8128.000000  2020.000000   1.000000e+07  2.360457e+06    14.000000   

         Postal_code  
count    7906.000000  
mean   503806.024285  
std    205061.544757  
min    110039.000000  
25%    391510.000000  
50%    524225.000000  
75%    680503.000000  
max    834008.000000

In [50]:
df.describe(include='object')

name    fuel seller_type transmission        owner  \
count                     7906    7906        7906         7906         7906   
unique                    1982       4           3            2            5   
top     Maruti Swift Dzire VDI  Diesel  Individual       Manual  First Owner   
freq                       129    4299        6563         6865         5215   

          mileage   engine max_power   City Sales_status  
count        7906     7906      7906   7906         7906  
unique        393      121       320     29            2  
top     18.9 kmpl  1248 CC    74 bhp  Delhi            N  
freq          225     1017       377    320         5906

## Q(1)  a) What are the characteristics of the cars listed in the used car dataset? Specifically, what is the average mileage and selling price of the cars? What is the range of prices? What is the interquartile range of prices? What is the most common fuel type and transmission type? Additionally, can we observe a significant difference in the mean selling price for cars sold by individual sellers compared to those sold by dealers in the used car dataset? b) Given that car has a selling price of over dollar 10,000, what is the probability that it has an automatic transmission? 

In [51]:
df['mileage'] = df['mileage'].str.replace(r'\b(kmpl|km/kg)\b', '', regex=True).astype(float)

# rest of the analysis
average_mileage = df['mileage'].mean()
average_price = df['selling_price'].mean()
price_range = df['selling_price'].max() - df['selling_price'].min()
price_interquartile_range = df['selling_price'].quantile(0.75) - df['selling_price'].quantile(0.25)
most_common_fuel_type = df['fuel'].mode()[0]
most_common_transmission_type = df['transmission'].mode()[0]




In [62]:

# Difference in mean selling price for individual sellers vs dealers
mean_price_individual_seller = df[df['seller_type'] == 'Individual']['selling_price'].mean()
mean_price_dealer = df[df['seller_type'] == 'Dealer']['selling_price'].mean()
price_difference = mean_price_individual_seller - mean_price_dealer

# Probability of having an automatic transmission given selling price > $10,000
cars_over_10k = df[df['selling_price'] > 10000]
probability_automatic_transmission = cars_over_10k[cars_over_10k['transmission'] == 'Automatic'].shape[0] / cars_over_10k.shape[0]

# Displaying the results
print("Characteristics of the cars:")
print("Average Mileage:", average_mileage)
print("Average Selling Price:", average_price)
print("Price Range:", price_range)
print("Interquartile Range of Prices:", price_interquartile_range)
print("Most Common Fuel Type:", most_common_fuel_type)
print("Most Common Transmission Type:", most_common_transmission_type)
print("\nDifference in Mean Selling Price for Individual Sellers vs Dealers:", price_difference)


#b)
print("\nProbability of having an automatic transmission given selling price > $10,000:", probability_automatic_transmission)

Characteristics of the cars:
Average Mileage: 19.4198608651657
Average Selling Price: 649813.720844928
Price Range: 9970001
Interquartile Range of Prices: 420000.0
Most Common Fuel Type: Diesel
Most Common Transmission Type: Manual

Difference in Mean Selling Price for Individual Sellers vs Dealers: -952203.803363302

Probability of having an automatic transmission given selling price > $10,000: 0.13167214773589678


## Q(2) a) In the context of analyzing the used car market, can we determine if there is a significant difference in prices of cars based on the fuel type they use? This information can help buyers make informed decisions when purchasing a used car and may also inform government policies aimed at promoting the use of certain types of fuel in vehicles. (frame hypothesis and draw inferences)

In [57]:


# Perform ANOVA test
anova_results = f_oneway(df[df['fuel'] == 'Petrol']['selling_price'],
                         df[df['fuel'] == 'Diesel']['selling_price'],
                         df[df['fuel'] == 'CNG']['selling_price'])

# Extract p-value
p_value = anova_results.pvalue

# Define significance level
alpha = 0.05

# Interpret the results
if p_value < alpha:
    print("Reject the null hypothesis. There is a significant difference in prices based on fuel type.")
else:
    print("Fail to reject the null hypothesis. There is no significant difference in prices based on fuel type.")

Reject the null hypothesis. There is a significant difference in prices based on fuel type.


Inference: This finding indicates that the type of fuel used in cars influences their selling prices in the used car market. Potential factors contributing to this difference could include variations in fuel efficiency, availability, environmental considerations, and consumer preferences. Understanding these price differences can assist buyers in making informed decisions when purchasing a used car and inform government policies aimed at promoting specific types of fuel in vehicles. Additionally, this information can be valuable for car dealerships in setting pricing strategies and managing inventory effectively.

## Q(3) a) In the used car dataset, there are some mileage values that are represented in units other than miles, such as kilometers. Convert these values into miles and check the mileage distribution. (Note: Convert mileage in kilometers to miles by multiplying by 0.621371.) 

In [59]:
#a)
#convert
df['mileage'] = df['mileage'].apply(lambda x: x * 0.621371 if 'km' in str(x) else x)

# Check the mileage distribution
mileage_distribution = df['mileage'].describe()

# Display the mileage distribution
print("Mileage Distribution:")
print(mileage_distribution)

Mileage Distribution:
count    7906.000000
mean       19.419861
std         4.036263
min         0.000000
25%        16.780000
50%        19.300000
75%        22.320000
max        42.000000
Name: mileage, dtype: float64


## b) You are a data analyst working for a car dealership company. Investigate whether there is a statistically significant difference in the distribution of the number of seats between cars with automatic and manual transmissions. Perform a hypothesis test to determine whether there is a significant difference in the number of seats for cars with automatic and manual transmissions.   Note: Use appropriate test to check if there is a significant difference in the means of the number of seats for the two groups. Determine the appropriate assumptions for the test and verify if they are satisfied. Calculate the test statistic and p-value to determine if there is a statistically significant difference between the two groups. Based on your analysis, provide insights to the management team about whether the number of seats is significantly different between cars with automatic and manual transmissions. This information will be useful in identifying which transmission type is more popular among customers and which features customers value the most. It will also help the company make informed decisions on inventory management and pricing strategies.

In [61]:
# Extract data for cars with automatic and manual transmissions
automatic_transmission = df[df['transmission'] == 'Automatic']['seats']
manual_transmission = df[df['transmission'] == 'Manual']['seats']

# Perform independent samples t-test
t_statistic, p_value = ttest_ind(automatic_transmission, manual_transmission)

# Display the results
print("Independent Samples t-test:")
print("Test Statistic:", t_statistic)
print("p-value:", p_value)

# Interpret the results
alpha = 0.05
if p_value < alpha:
    print("Reject the null hypothesis. There is a statistically significant difference in the distribution of the number of seats between cars with automatic and manual transmissions.")
else:
    print("Fail to reject the null hypothesis. There is no statistically significant difference in the distribution of the number of seats between cars with automatic and manual transmissions.")

Independent Samples t-test:
Test Statistic: -6.482469584385468
p-value: 9.564849249329582e-11
Reject the null hypothesis. There is a statistically significant difference in the distribution of the number of seats between cars with automatic and manual transmissions.


Test Statistic: The test statistic measures the difference between the means of the two groups (cars with automatic and manual transmissions) relative to the variability within the groups. In this case, the test statistic is -6.48.
p-value: The p-value is a measure of the probability of observing the data or more extreme data under the assumption that the null hypothesis is true. A smaller p-value indicates stronger evidence against the null hypothesis. In this case, the p-value is approximately 9.56e-11, which is extremely small.

Inference: With a p-value much smaller than the significance level (typically 0.05), we reject the null hypothesis. Therefore, we conclude that there is a statistically significant difference in the distribution of the number of seats between cars with automatic and manual transmissions.
Implications: This finding suggests that the choice of transmission type (automatic or manual) influences the number of seats in cars. Customers may have different preferences regarding the number of seats depending on the transmission type. For example, cars with automatic transmissions may be preferred for convenience and comfort, while cars with manual transmissions may appeal to enthusiasts seeking performance or a different driving experience.